### PR TITLE
Sync the Flow tests

### PR DESCRIPTION
**Pull request by lydell**
Tip: Take a look at each commit one by one.

I’ve glanced through the updated and new snapshots, and not found anything obviously strange.

### CODE REVIEW

File: tests/flow/getters_and_setters_enabled/jsfmt.spec.js [hunk 1]
```diff
@@ -1 +1,5 @@
-run_spec(__dirname, null, ["babylon"]);
+run_spec(__dirname);
+
+// TODO: These tests used to be run in babylon as well, but currently babylon
+// fails to parse them.
+// run_spec(__dirname, null, ["babylon"]);
```
Comment (lydell) [inline tests/flow/getters_and_setters_enabled/jsfmt.spec.js]: I had to disable babylon here for now.

```
Summary of all failing tests
 FAIL  tests/flow/getters_and_setters_enabled/jsfmt.spec.js
  ● declare_class.js - babylon-verify

    SyntaxError: Unexpected token, expected : (8:6)
       6 | 
       7 | declare class Foo {
    >  8 |   get goodGetterWithAnnotation(): number;
         |       ^
       9 |   set goodSetterWithAnnotation(x: number): void;
      10 | 
      11 |   get propWithMatchingGetterAndSetter(): number;
      
  ● object_type.js - babylon-verify

    SyntaxError: Unexpected token, expected : (12:6)
      10 | 
      11 | type T = {
    > 12 |   get goodGetterWithAnnotation(): number,
         |       ^
      13 |   set goodSetterWithAnnotation(x: number): void,
      14 | 
      15 |   get propWithMatchingGetterAndSetter(): number,
```

File: tests/flow/annot/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -163,38 +131,6 @@ function foobar(n: ?number): number | null | void {
 function barfoo(n: number | null | void): ?number {
   return n;
 }
-
-type Banana = {
```
Comment (vjeux) [inline tests/flow/annot/__snapshots__/jsfmt.spec.js.snap]: Those are tests that we manually added in the wrong place. Would be nice to restore them back outside of the flow folder.

File: tests/flow/getters_and_setters_enabled/jsfmt.spec.js (hunk 1 above)
Comment (vjeux) [inline tests/flow/getters_and_setters_enabled/jsfmt.spec.js]: cc @hzoo

File: tests/flow/annot/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (lydell) [inline tests/flow/annot/__snapshots__/jsfmt.spec.js.snap]: Check out the first commit ;)

File: tests/flow_function_parentheses/test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+type Banana = {
```
Comment (lydell) [inline tests/flow_function_parentheses/test.js]: Here it is, @vjeux

Comment (vjeux) [inline tests/flow_function_parentheses/test.js]: Yay, thanks!

**Comment by vjeux** [comment]
Thanks for doing that!

File: tests/flow/getters_and_setters_enabled/jsfmt.spec.js (hunk 1 above)
Comment (existentialism) [inline tests/flow/getters_and_setters_enabled/jsfmt.spec.js]: Will be fixed in https://github.com/babel/babylon/pull/224

Comment (vjeux) [inline tests/flow/getters_and_setters_enabled/jsfmt.spec.js]: Yay, thanks!!

**Pull request merged**